OMCompiler/Compiler/NFFrontEnd/NFExpression.mo

```diff
@@ -6457,5 +6480,22 @@ public
     end match;
   end replaceLiteral;
 
+  function replaceResizableParameter
+    input output Expression exp;
+  algorithm
+    exp := match exp
+      local
+        Integer v;
+        Option<Integer> value;
+      case Expression.CREF() guard(Expression.variability(exp) == Variability.NON_STRUCTURAL_PARAMETER) algorithm
+        value := match InstNode.getBindingExpOpt(ComponentRef.node(exp.cref))
+          case SOME(Expression.INTEGER(v)) then SOME(v);
+          case SOME(Expression.SUBSCRIPTED_EXP(exp = Expression.INTEGER(v))) then SOME(v);
+          else NONE();
+        end match;
+      then if Util.isSome(value) then Expression.INTEGER(Util.getOption(value)) else exp;
```

Comment:
Unless you expect to add many more cases here I'd just return the expression directly instead of wrapping the integer in an option, it's a lot of extra work just to avoid repeating `Expression.INTEGER` otherwise.